ocaml/xapi/xapi_pif.ml

```diff
@@ -51,11 +106,21 @@ let refresh_internal ~__context ~self =
       pif.API.pIF_network
   in
   let bridge = Db.Network.get_bridge ~__context ~self:network in
+  (* Pif device name maybe change. Look up device_to_position table to get the
+     new device name. *)
+  let pif_device_name =
+    Option.bind (n_of_xenbrn_opt bridge) (fun position ->
+        List.find_map
+          (fun (name, pos) -> if pos = position then Some name else None)
+          interface_tables.device_to_position_table
+    )
+    |> Option.value ~default:pif.API.pIF_device
```

Comment:
We need a log to understand what is happening here. The case is one of:
1. the device of the PIF disappears from the system
2. a bug in device sorting logic.
With regarding the interface-rename, the two can not be distinguished; while with regarding the ordering in networkd, they can.
If it is the case 1, the `Net.Interface.get_mac` will raise exception. If it is the case 2, we can't hide the bug.

Reply:
We can't find a device from the order but `get_mac` can return its mac.

Reply:
It could be
1. a `ibft` interface, 
2. unplugged interface, or
3. missed in `get_interface_positions` due to a bug.